src/ast/operation_visitor.rs

```diff
@@ -0,0 +1,689 @@
+            visitor.leave_scalar_value(context, v);
+        }
+        Value::Int(v) => {
+            visitor.enter_scalar_value(context, v);
+            visitor.leave_scalar_value(context, v);
+        }
+        Value::Null => {
+            visitor.enter_null_value(context, ());
+            visitor.leave_null_value(context, ());
+        }
+        Value::String(v) => {
+            visitor.enter_scalar_value(context, v);
+            visitor.leave_scalar_value(context, v);
+        }
+        Value::Enum(v) => {
+            visitor.enter_enum_value(context, v.clone());
+            visitor.leave_enum_value(context, v.clone());
+        }
+        Value::List(v) => {
+            visitor.enter_list_value(context, v.clone());
+
+            let input_type = context.current_input_type_literal().and_then(|t| match t {
+                Type::ListType(inner_type) => Some(inner_type.as_ref().clone()),
+                _ => None,
+            });
+
+            context.with_input_type(input_type, |context| {
+                for item in v {
+                    visit_input_value(visitor, item, context)
+                }
+            });
+
+            visitor.leave_list_value(context, v.clone());
+        }
+        Value::Object(v) => {
+            visitor.enter_object_value(context, v.clone());
+
+            for (sub_key, sub_value) in v.iter() {
+                let input_type = context
+                    .current_input_type_literal()
+                    .and_then(|v| context.schema.type_by_name(&v.named_type()))
+                    .and_then(|v| v.input_field_by_name(&sub_key))
+                    .and_then(|v| Some(v.value_type));
+
+                context.with_input_type(input_type, |context| {
+                    let param = &(sub_key.clone(), sub_value.clone());
+                    visitor.enter_object_field(context, param);
+                    visit_input_value(visitor, sub_value, context);
+                    visitor.leave_object_field(context, param);
+                });
+            }
+
+            visitor.leave_object_value(context, v.clone());
+        }
+        Value::Variable(v) => {
+            visitor.enter_variable_value(context, v.clone());
+            visitor.leave_variable_value(context, v.clone());
+        }
+    }
+}
+
+fn visit_variable_definitions<'a, Visitor, UserContext>(
+    visitor: &mut Visitor,
+    variables: &[VariableDefinition],
+    context: &mut OperationVisitorContext<'a, UserContext>,
+) where
+    Visitor: OperationVisitor<'a, UserContext>,
+{
+    for variable in variables {
+        context.with_input_type(Some(variable.var_type.clone()), |context| {
+            visitor.enter_variable_definition(context, variable);
+
+            if let Some(default_value) = &variable.default_value {
+                visit_input_value(visitor, &default_value, context);
+            }
+
+            // DOTAN: We should visit the directives as well here, but it's extracted in graphql_parser.
```

Comment:
technically yes we want to visit directives here too but `graphql_parser` doesn't support directives on variable definitions yet